From 16bcd760f5584256001b0b8c2806a645bffc6bb8 Mon Sep 17 00:00:00 2001
From: Hitoshi Kojima <hitoshi.kojima.wt@ps.hitachi-solutions.com>
Date: Fri, 18 Nov 2016 11:37:35 +0900
Subject: [PATCH 1/2] Revert "v4l2object: Compensate for the interpretation
 difference of colorspace definitions between VIN and
 GStreamer"

This reverts commit 93f8898a627a239b2561fec6b6c55a978b692a38.
---
 sys/v4l2/gstv4l2object.c |    8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/sys/v4l2/gstv4l2object.c b/sys/v4l2/gstv4l2object.c
index 1c6ae73..1cd5b92 100644
--- a/sys/v4l2/gstv4l2object.c
+++ b/sys/v4l2/gstv4l2object.c
@@ -1431,9 +1431,7 @@ gst_v4l2_object_v4l2fourcc_to_structure (guint32 fourcc)
           format = GST_VIDEO_FORMAT_BGR;
           break;
         case V4L2_PIX_FMT_RGB32:
-          /* Compensate for the interpretation difference of
-             colorspace definition between VIN and GStreamer */
-          format = GST_VIDEO_FORMAT_BGRx;
+          format = GST_VIDEO_FORMAT_RGBx;
           break;
         case V4L2_PIX_FMT_BGR32:
           format = GST_VIDEO_FORMAT_BGRx;
@@ -1644,9 +1642,7 @@ gst_v4l2_object_get_caps_info (GstV4l2Object * v4l2object, GstCaps * caps,
         break;
       case GST_VIDEO_FORMAT_BGRx:
       case GST_VIDEO_FORMAT_BGRA:
-        /* Compensate for the interpretation difference of colorspace definition
-           between VIN and GStreamer */
-        fourcc = V4L2_PIX_FMT_RGB32;
+        fourcc = V4L2_PIX_FMT_BGR32;
         break;
       case GST_VIDEO_FORMAT_GRAY8:
         fourcc = V4L2_PIX_FMT_GREY;
-- 
1.7.9.5

